js/modules/types/attachment.js

```diff
@@ -53,76 +36,14 @@ exports.isValid = (rawAttachment) => {
     return false;
   }
 
-  return isString(rawAttachment.contentType) &&
-    isString(rawAttachment.fileName);
-};
-
-// Middleware
-// type UpgradeStep = Attachment -> Promise Attachment
-
-// SchemaVersion -> UpgradeStep -> UpgradeStep
-exports.withSchemaVersion = (schemaVersion, upgrade) => {
-  if (!isNumber(schemaVersion)) {
-    throw new TypeError('`schemaVersion` must be a number');
-  }
-  if (!isFunction(upgrade)) {
-    throw new TypeError('`upgrade` must be a function');
-  }
-
-  return async (attachment) => {
-    if (!exports.isValid(attachment)) {
-      console.log('Attachment.withSchemaVersion: Invalid input attachment:', attachment);
-      return attachment;
-    }
-
-    const isAlreadyUpgraded = attachment.schemaVersion >= schemaVersion;
-    if (isAlreadyUpgraded) {
-      return attachment;
-    }
-
-    const expectedVersion = schemaVersion - 1;
-    const isUnversioned = isUndefined(attachment.schemaVersion);
-    const hasExpectedVersion = isUnversioned ||
-      attachment.schemaVersion === expectedVersion;
-    if (!hasExpectedVersion) {
-      console.log(
-        'WARNING: Attachment.withSchemaVersion: Unexpected version:' +
-        ` Expected attachment to have version ${expectedVersion},` +
-        ` but got ${attachment.schemaVersion}.`,
-        attachment
-      );
-      return attachment;
-    }
-
-    let upgradedAttachment;
-    try {
-      upgradedAttachment = await upgrade(attachment);
-    } catch (error) {
-      console.log(
-        'Attachment.withSchemaVersion: error:',
-        error && error.stack ? error.stack : error
-      );
-      return attachment;
-    }
-
-    if (!exports.isValid(upgradedAttachment)) {
-      console.log(
-        'Attachment.withSchemaVersion: Invalid upgraded attachment:',
-        upgradedAttachment
-      );
-      return attachment;
-    }
-
-    return Object.assign(
-      {},
-      upgradedAttachment,
-      { schemaVersion }
-    );
-  };
+  const hasValidContentType = isString(rawAttachment.contentType);
+  const hasValidFileName =
+    isString(rawAttachment.fileName) || rawAttachment.fileName === null;
+  return hasValidContentType && hasValidFileName;
 };
 
 // Upgrade steps
-const autoOrientJPEG = async (attachment) => {
+exports.autoOrientJPEG = async (attachment) => {
```

Comment:
I know that sprinkling exports all through the file is the future, given the new `export` keyword. But for now I'd really like us to have all the exports in a given file in one place. With function hoisting you can even have it at the top of the file. As an example, check `backup.js` in my backup prep PR.

Reply:
Sure, I am happy to make that change if you feel strongly. But if we were to adopt `export`, we’d go back again 😉 Good news: PureScript does it explicitly on top as well https://github.com/truqu/purescript-indexedDB/blob/ef6b6498c356454f85b4c549ed4fd98b3d171944/src/Database/IndexedDB/IDBDatabase.purs#L3-L25

Reply:
As discussed, will consider this for the future ✨